cpp/include/cuvs/neighbors/graph_build_types.hpp

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
cuVS isn't header-only, so the header files are very lightweight. Instead of introducing new headers for structs and enums, can we just reuse common.hpp?

Reply:
`common.hpp` has the base `index_params` ([link](https://github.com/rapidsai/cuvs/blob/e8dbb887cf3a9646ed20dcf4b749528ecd3e15e6/cpp/include/cuvs/neighbors/common.hpp#L90-L95)), so declaring `graph_build_params` in that file results in a circular dependency (because we make use of `ivf_pq::index_params` and `nn_descent::index_params` inside `graph_build_params)